### PR TITLE
Renamed SequenceContext to ActionContext

### PR DESCRIPTION
**Pull request by Stevenic**
- Renamed class and file.
- Cleaned up various variable names.
- Made the context passed into actions a standard DC.
- Updated actions to work against their parent ActionContext for things like sequence manipulation.

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/DeleteProperties.cs [hunk 1]
```diff
@@ -65,23 +65,15 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
                 return await dc.EndDialogAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
             }
 
-            // Ensure planning context
-            if (dc is SequenceContext planning)
+            if (this.Properties?.Any() == true)
             {
-                if (this.Properties?.Any() == true)
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/DeleteProperties.cs]: Not used any more?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/ActionContext.cs [hunk 2]
```diff
@@ -120,7 +120,7 @@ public async Task<bool> ApplyChangesAsync(CancellationToken cancellationToken =
             return false;
         }
 
-        public SequenceContext InsertActions(List<ActionState> actions)
+        public ActionContext InsertActions(List<ActionState> actions)
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/ActionContext.cs]: >InsertActions [](start = 29, length = 13)

Can you add method doc comments for all of these?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/ActionContext.cs [hunk 3]
```diff
@@ -9,11 +9,11 @@
 
 namespace Microsoft.Bot.Builder.Dialogs.Adaptive
 {
-    public class SequenceContext : DialogContext
+    public class ActionContext : DialogContext
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/ActionContext.cs]: >ActionContext [](start = 17, length = 13)

Doc comment please?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs [hunk 4]
```diff
@@ -647,7 +645,7 @@ protected async Task<RecognizerResult> OnRecognize(SequenceContext sequenceConte
 
         // This function goes through the ambiguity queues and emits events if present.
         // In order ClearProperties, AssignEntity, ChooseProperties, ChooseEntity, EndOfActions.
-        private async Task<bool> ProcessQueuesAsync(SequenceContext sequenceContext, CancellationToken cancellationToken)
+        private async Task<bool> ProcessQueuesAsync(ActionContext sequenceContext, CancellationToken cancellationToken)
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs]: >sequenceContext [](start = 66, length = 15)

Can rename all of the sequenceContext variable names to actionContext?

**Pull request merged**